File: HealthKitSample/AddDataPointViewController.m
//
//  AddDataPointViewController.m
//  HealthKitSample
//
//  Created by Eric Mansfield on 6/4/15.
//  Copyright (c) 2015 Eric Mansfield. All rights reserved.
//

#import "AddDataPointViewController.h"

@import HealthKit;

@interface AddDataPointViewController ()

@property (strong, nonatomic) HKHealthStore *healthStore;

@end

@implementation AddDataPointViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.healthStore = [[HKHealthStore alloc] init];

    // update placeholder text
    self.dataTextField.placeholder = self.preferredUnit.unitString;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


- (IBAction)cancelTapped:(id)sender {
    [self.navigationController popViewControllerAnimated:YES];
}

- (IBAction)saveTapped:(id)sender {
    double value = self.dataTextField.text.doubleValue;
    HKQuantity *quantity = [HKQuantity quantityWithUnit:self.preferredUnit doubleValue:value];
    NSDate *now = [NSDate date];
    
    HKQuantitySample *quantitySample = [HKQuantitySample quantitySampleWithType:self.quantityType quantity:quantity startDate:now endDate:now];
    
    [self.healthStore saveObject:quantitySample withCompletion:^(BOOL success, NSError *error) {
        if (!success) {
            NSLog(@"An error occured %@.", error);
            abort();
        }
        else {
            NSLog(@"Data saved to HealthKit!!!!.");
            [[NSOperationQueue mainQueue] addOperationWithBlock:^{
                [self.navigationController popViewControllerAnimated:YES];
            }];
        }
    }];
}

- (IBAction)editingChanged:(id)sender {
    self.saveButton.enabled = [self.dataTextField.text stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]].length > 0;
}

@end

File: HealthKitSample-Swift/HealthKitSample-Swift/AddDataPointViewController.swift
//
//  AddDataPointViewController.swift
//  HealthKitSample-Swift
//
//  Created by Eric Mansfield on 6/8/15.
//  Copyright (c) 2015 Eric Mansfield. All rights reserved.
//

import HealthKit
import UIKit

class AddDataPointViewController: UITableViewController {
    
    @IBOutlet weak var saveButton: UIBarButtonItem!
    @IBOutlet weak var unitsLabel: UILabel!
    @IBOutlet weak var dataTextField: UITextField!
    
    var quantityType: HKQuantityType!
    var preferredUnit: HKUnit!
    var healthStore = HKHealthStore()
    

    override func viewDidLoad() {
        super.viewDidLoad()

        // update placeholder text
        self.unitsLabel.text = self.preferredUnit.unitString;
        
        self.dataTextField.becomeFirstResponder()

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    // MARK: - IBActions
    @IBAction func editingChanged(sender: AnyObject) {
        let textField = sender as! UITextField
        let value = textField.text!.stringByTrimmingCharactersInSet(NSCharacterSet.whitespaceCharacterSet())
        
        if (value.characters.count > 0) {
            self.saveButton.enabled = true
        }
        else {
            self.saveButton.enabled = false
        }
    }
    
    @IBAction func saveTapped(sender: AnyObject) {
        let doubleValue = (self.dataTextField.text! as NSString).doubleValue
        let quantity = HKQuantity(unit: self.preferredUnit, doubleValue: doubleValue)
        
        let date = NSDate()
        
        let quantitySample = HKQuantitySample(type: self.quantityType, quantity: quantity, startDate: date, endDate: date)
        
        
        self.healthStore.saveObject(quantitySample) { (success, error) -> Void in
            if (success) {
                NSLog("Saved to healthkiit.....")
                
                NSOperationQueue.mainQueue().addOperationWithBlock({ () -> Void in
                    self.navigationController?.popViewControllerAnimated(true)
                })
            }
        }
    }
    
    @IBAction func cancelTapped(sender: AnyObject) {
        NSLog("Cancel tapped....")
        self.navigationController?.popViewControllerAnimated(true)
    }
}
